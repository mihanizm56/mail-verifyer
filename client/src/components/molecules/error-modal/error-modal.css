.error-modal {
    display: flex;
    flex-direction: column;
    border: 1px solid red;
    border-radius: 20px;
    padding-top: 25px;
    width: 700px;
    min-height: 500px;
}

.error-modal__content {
    padding-top: 50px;
    min-height: 100px;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
}

.error-modal__button {
    padding: 100px 0 25px 0;
    display: flex;
    justify-content: center;
}

@media screen and (max-width: 1199px) and (min-width: 768px) {
    .error-modal {
        min-height: 500px;
        width: 70%;
    }
    .error-modal__content {
        min-height: 100px;
    }
}

@media screen and (max-width: 767px) {
    .error-modal {
        min-height: 500px;
        width: 90%;
    }
    .error-modal__content {
        padding: 50px 10px 0 10px;
    }
}