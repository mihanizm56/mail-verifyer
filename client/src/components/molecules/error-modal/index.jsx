import React from "react";
import { useTranslation } from "react-i18next";
import errorIcon from "../../../assets/icons/icon-error.svg";
import { SVGComponent, OpenSansText, LinkButton } from "../../../components";
import { errorCreator } from "../../../utils/helpers/error-creator";
import "./error-modal.css";

export const ErrorModal = ({ error, handleButtonClick }) => {
	const errorText = errorCreator(error);
	const { t: translate } = useTranslation();

	return (
		<div className="error-modal">
			<SVGComponent path={errorIcon} />
			<div className="error-modal__content">
				<OpenSansText text={translate("errors.error-type-label")} />
				<OpenSansText text={errorText} />
			</div>
			<div className="error-modal__button">
				<LinkButton to="/send" text={translate("buttons.send-new")} handleButtonClick={handleButtonClick} />
			</div>
		</div>
	);
};

ErrorModal.defaultProps = {
	handleButtonClick: () => console.log("default handleButtonClick"),
};
