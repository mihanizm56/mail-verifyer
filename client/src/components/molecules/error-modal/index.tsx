import React, { memo } from "react";
import { useTranslation } from "react-i18next";
import errorIcon from "../../../assets/icons/icon-error.svg";
import { SVGIcon, OpenSansText, LinkButton } from "../..";
import { errorCreator } from "../../../utils/helpers/error-creator";
import "./error-modal.css";

type ErrorModalProps = {
	error?: string,
	handleButtonClick?: () => void,
};

export const ErrorModal = memo(({ error, handleButtonClick }: ErrorModalProps) => {
	const errorText = errorCreator(error);
	const { t: translate } = useTranslation();

	return (
		<div className="error-modal">
			<SVGIcon path={errorIcon} />
			<div className="error-modal__content">
				<OpenSansText text={translate("errors.error-type-label")} />
				<OpenSansText text={errorText} />
			</div>
			<div className="error-modal__button">
				<LinkButton to="/send" text={translate("buttons.send-new")} handleButtonClick={handleButtonClick} />
			</div>
		</div>
	);
});
