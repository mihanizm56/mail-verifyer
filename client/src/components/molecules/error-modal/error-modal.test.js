import React, { createElement } from "react";
import { shallow } from "enzyme";
import { ErrorModal } from "../error-modal";

describe("test ErrorModal", () => {
	beforeEach(() => {
		jest.resetModules();
		jest.clearAllMocks();
	});

	// SVGIconMock
	// OpenSansTextMock
	// OpenSansTextMock
	// LinkButtonMock

	describe("test render contract", () => {
		it("render snap with all props", () => {
			const props = {
				error: "test error",
				handleButtonClick: jest.fn(),
			};
			const component = shallow(<ErrorModal {...props} />);

			expect(component).toMatchSnapshot();
		});
		it("render SVGIcon component", () => {
			const props = {
				error: "test error",
				handleButtonClick: jest.fn(),
			};
			const component = shallow(<ErrorModal {...props} />);
			const SVGIcon = component.find("SVGIconMock");

			expect(SVGIcon.length).toEqual(1);
		});
		it("render OpenSansText two components", () => {
			const props = {
				error: "test error",
				handleButtonClick: jest.fn(),
			};
			const component = shallow(<ErrorModal {...props} />);
			const OpenSansTexts = component.find("OpenSansTextMock");

			expect(OpenSansTexts.length).toEqual(2);
		});
		it("render LinkButton two components", () => {
			const props = {
				error: "test error",
				handleButtonClick: jest.fn(),
			};
			const component = shallow(<ErrorModal {...props} />);
			const LinkButton = component.find("LinkButtonMock");

			expect(LinkButton.length).toEqual(1);
		});
	});
	describe("test props contract", () => {
		it("set path prop to the SVGIcon", () => {
			const props = {
				error: "test error",
				handleButtonClick: jest.fn(),
			};
			const component = shallow(<ErrorModal {...props} />);
			const SVGIcon = component.find("SVGIconMock");
			const iconPath = SVGIcon.props().path;

			expect(iconPath).toBeDefined();
		});
		it("set text prop to the first OpenSansText", () => {
			const props = {
				error: "test error",
				handleButtonClick: jest.fn(),
			};
			const component = shallow(<ErrorModal {...props} />);
			const OpenSansText = component.find("OpenSansTextMock").first();
			const textValue = OpenSansText.prop("text");

			expect(textValue).toEqual("errors.error-type-label");
		});
		it("set text prop to the last OpenSansText", () => {
			const props = {
				error: "test error",
				handleButtonClick: jest.fn(),
			};
			const component = shallow(<ErrorModal {...props} />);
			const OpenSansText = component.find("OpenSansTextMock").last();
			const textValue = OpenSansText.prop("text");

			expect(textValue).toEqual("errors.server");
		});
		it("set handleButtonClick prop to the last LinkButton", () => {
			const props = {
				error: "test error",
				handleButtonClick: jest.fn(),
			};
			const component = shallow(<ErrorModal {...props} />);
			const LinkButton = component.find("LinkButtonMock");
			const handleButtonClick = LinkButton.prop("handleButtonClick");

			expect(handleButtonClick).toEqual(props.handleButtonClick);
		});
		it("set text prop to the last LinkButton", () => {
			const props = {
				error: "test error",
				handleButtonClick: jest.fn(),
			};
			const component = shallow(<ErrorModal {...props} />);
			const LinkButton = component.find("LinkButtonMock");
			const text = LinkButton.prop("text");

			expect(text).toEqual("buttons.send-new");
		});
		it("set to prop (path) to the last LinkButton", () => {
			const props = {
				error: "test error",
				handleButtonClick: jest.fn(),
			};
			const component = shallow(<ErrorModal {...props} />);
			const LinkButton = component.find("LinkButtonMock");
			const path = LinkButton.prop("to");

			expect(path).toEqual("/send");
		});
	});
});
