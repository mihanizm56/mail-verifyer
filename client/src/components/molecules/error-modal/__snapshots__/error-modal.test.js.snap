// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ErrorModal test render contract render snap with all props 1`] = `
<div
  className="error-modal"
>
  <SVGIconMock
    path="icon-error.svg"
  />
  <div
    className="error-modal__content"
  >
    <OpenSansTextMock
      text="errors.error-type-label"
    />
    <OpenSansTextMock
      text="errors.server"
    />
  </div>
  <div
    className="error-modal__button"
  >
    <LinkButtonMock
      handleButtonClick={[MockFunction]}
      text="buttons.send-new"
      to="/send"
    />
  </div>
</div>
`;
